kernel/src/callback.rs

```diff
@@ -1,30 +1,52 @@
 //! Data structure for storing a callback to userspace or kernelspace.
 
+use core::fmt;
 use core::ptr::NonNull;
-use process;
+
+use process::{self, Process};
 
 /// Userspace app identifier.
-#[derive(Clone, Copy, Eq, PartialEq, Debug)]
+#[derive(Clone, Copy)]
 pub struct AppId {
+    crate process: &'static Process<'static>,
```

Comment:
I think this is a dangerous change, and that it should be reverted to storing a reference to the `Kernel`